File: pattern_divide_and_conquer/search_algorithms/linear_search.js
function LinearSearch(arr, val) {
    for (let i = 0; i < arr.length; i++) {
        if (val === arr[i] ) {
            return i;
        }
    }
    return -1
}

console.log(LinearSearch([1, 2, 3, 4, 5, 6, 7], 6))

File: pattern_divide_and_conquer/search_algorithms/linear_search.py
def linear_search(arr, low, high, key):
    """
    :param arr: array of values
    :param low: the lower bound
    :param high: the upper bound
    :param key: the key from which to search
    :return:
    """
    if high < low:
        return 'NOT_FOUND'
    if arr[low] == key:
        return low
    return linear_search(arr, low + 1, high, key)